src/modules/partition/core/PartitionLayout.cpp

```diff
@@ -164,60 +164,118 @@ PartitionLayout::execute( Device* dev,
                           const PartitionRole& role )
 {
     QList< Partition* > partList;
+    QMap< const PartitionLayout::PartitionEntry *, qint64 > partSizeMap;
     qint64 minSize, maxSize, end;
     qint64 totalSize = lastSector - firstSector + 1;
     qint64 availableSize = totalSize;
+    qint64 minPartitionSize = 0;
 
-    // TODO: Refine partition sizes to make sure there is room for every partition
-    // Use a default (200-500M ?) minimum size for partition without minSize
-
+    // Let's check if we have enough space for each partSize
     foreach ( const PartitionLayout::PartitionEntry& part, m_partLayout )
```

Comment:
Though this isn't your code, Calamares now uses range-style for, so `for( const PartitionLayout::PartitionsEntry& part : m_partLayout )`. That convention has changed since this code was first written, and now I see it's not updated consistently everywhere. The same applies to your other foreach()es, could you re-do them in range-style?

Reply:
changed